common/test/TestDevice.cs

```diff
@@ -25,26 +21,23 @@ public enum ConnectionStringAuthScope
         Device
     }
 
-    public class TestDevice
+    public class TestDevice : IDisposable
```

Comment:
Has a SemaphoreSlim which is IDisposable, so let's make this class disposable to stay cleaned up, even during a test run.